cloudant-client/src/main/java/com/cloudant/client/api/Database.java

```diff
@@ -426,16 +535,41 @@ public void createIndex(String indexDefinition) {
     }
 
     /**
-     * Delete an index
+     * List the indexes in the database. The returned object allows for listing indexes by type.
+     *
+     * @return indexes object with methods for getting indexes of a particular type
+     */
+    public Indexes listIndexes() {
+        URI uri = new DatabaseURIHelper(db.getDBUri()).path("_index").build();
+        return client.couchDbClient.get(uri, Indexes.class);
+    }
+
+    /**
+     * Delete a JSON index
      *
      * @param indexName   name of the index
      * @param designDocId ID of the design doc
      */
     public void deleteIndex(String indexName, String designDocId) {
+        deleteIndex(indexName, designDocId, "json");
+    }
+
+    /**
+     * Delete an index with the specified name and type in the given design document.
+     *
+     * @param indexName   name of the index
+     * @param designDocId ID of the design doc (the _design prefix will be added if not present)
+     * @param type        type of the index, defaults to "json"
+     */
+    public void deleteIndex(String indexName, String designDocId, String type) {
```

Comment:
This method shouldn't default to "json" because that's not obvious behaviour.

We should probably have symbolic constants for "json" and "text" and document that these are the known valid options (at this time) in the javadoc here.